device-code-flow/app.ps1

```diff
@@ -0,0 +1,63 @@
+# A PowerShell application that demonstrates how to use the
+# Device Code flow to make an API call to Microsoft Graph.
+
+# Import the MSAL module
+Import-Module -Name Microsoft.Identity.Client
+
+# Assemblies required by DeviceCodeHelper
+[System.Collections.Generic.List[string]] $RequiredAssemblies = New-Object System.Collections.Generic.List[string]
+$RequiredAssemblies.Add('System.Console.dll')
+$RequiredAssemblies.Add('Microsoft.Identity.Client')
+
+# DeviceCodeHelper, written in C#, will display a message on the console
+# instructing the user how to authenticate via their device.
+# AcquireTokenWithDeviceCode() will poll the server after firing the
+# device code callback to look for a successful login with the provided code.
+Add-Type -TypeDefinition @"
+using System;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+
+public static class DeviceCodeHelper
+{
+    public static Func<DeviceCodeResult,Task> GetDeviceCodeResultCallback()
+    {
+        return deviceCodeResult =>
+        {
+            Console.WriteLine(deviceCodeResult.Message);
+            return Task.FromResult(0);
+        };
+    }
+}
+"@ -ReferencedAssemblies $RequiredAssemblies -IgnoreWarnings -WarningAction SilentlyContinue
+
+# 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+$ClientId = ""
+
+# 'Directory (tenant) ID' of app registration in Azure portal - this value is a GUID
+$TenantId = ""
+
+# Scope permission for Graph
+[string[]] $Scope = "User.Read"
```

Comment:
Any chance you can inline this on line 48?  We've inlined this in most of the other ones so that it's easy to see (when taken as a local snippet), what scope is being asked for.

Reply:
Yep - fix coming

Reply:
updated

Reply:
looks good!